src/library/scala/collection/immutable/Map.scala

```diff
@@ -108,6 +108,28 @@ object Map extends ImmutableMapFactory[Map] {
     def iterator: Iterator[(Any, Nothing)] = Iterator.empty
     override def updated [V1] (key: Any, value: V1): Map[Any, V1] = new Map1(key, value)
     def + [V1](kv: (Any, V1)): Map[Any, V1] = updated(kv._1, kv._2)
+    override def ++[V1 >: Nothing](xs: GenTraversableOnce[(Any, V1)]): Map[Any, V1] = addImpl(xs)
+    override def ++[B >: (Any, Nothing), That](that: GenTraversableOnce[B])(implicit bf: CanBuildFrom[Map[Any, Nothing], B, That]): That = super.++(that)
+    private def addImpl[B >: (Any, Nothing), That](that: GenTraversableOnce[B])(implicit bf: CanBuildFrom[Map[Any, Nothing], B, That]): That = {
```

Comment:
It looks like this is only called from the first `++` above where the cbf instance is statically known.

Reply:
doh - changed to be similar to HashMap `++` and `addImpl`